src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/QualityGuidelines/UseLiteralsWhereAppropriateTests.cs

```diff
@@ -169,6 +169,26 @@ End Class
             await vbTest.RunAsync();
         }
 
+        [Fact]
+        public async Task CA1802_CSharp_IntPtr_UIntPtr_NoDiagnostic()
+        {
+            await new VerifyCS.Test
+            {
+                LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.CSharp9,
+                ReferenceAssemblies = ReferenceAssemblies.Net.Net50,
+                TestCode = @"
+using System;
+
+public class Class1
+{
+	internal static readonly IntPtr field1 = (nint)0;
+	internal static readonly UIntPtr field2 = (nuint)0;
+    internal static readonly nint field3 = (nint)0;
+	internal static readonly nuint field4 = (nuint)0;
```

Comment:
I expect these to have a diagnostic. They can be made as constants.

See [SharpLab](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxAN0QSwGYE8A+AAgEwCMAsAFCEDMABCXQMIA20UFlA3lQJCYA7eAgEQWdAMYB7AbDoBJIQAUYCOtkxwWAE1J0AvHQAUAwTACUABgDcfM4lHjpsmHQCqimCrUat24gbGAgCuZla2lHRRdPYiYpIycqZC6po69IYmYTZ2Qg7xzkmhKb46ACyBJsUWOZQAvkA).

Reply:
Yes but they are considered as `IntPtr` and `UIntPtr` by roslyn so I thought it was better to also exclude them.

Reply:
The reason in the original issue was that there is no way to fix the violation, as they can't be made constants.

But if in any case they can be made constants, the diagnostic should be reported. Tagging @sharwell to confirm in case I'm wrong.

Reply:
Agree with @Youssef1313 (these two lines should still have a diagnostic)